source/yats/include/yats/task_container.h

```diff
@@ -47,12 +47,19 @@ class task_container : public abstract_task_container
 
 protected:
     template <size_t... index, typename T = typename helper::output_type>
-    std::enable_if_t<!std::is_same<T, void>::value> invoke(std::integer_sequence<size_t, index...>)
+    std::enable_if_t<!std::is_same<T, void>::value && is_tuple_v<T>> invoke(std::integer_sequence<size_t, index...>)
```

Comment:
We do not need to check for T != void because then is_tuple_v<T> returns false anyway